mars/dataframe/tests/test_core.py

```diff
@@ -320,3 +320,92 @@ def test_key_value(setup):
+    pd_series[::2] = np.nan
+    expected = pd_series[3:18].dropna()
+
+    series = Series(pd_series, chunk_size=5)
+    result = series[series.between(series[3], series[17])]
+    result = result.execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+    result = series[series.between(series[3], series[17], inclusive="neither")]
+    result = result.execute().fetch()
+    expected = pd_series[5:16].dropna()
+    pd.testing.assert_series_equal(result, expected)
+
+
+# def test_between_period_values(setup):
+#     pd_series = pd.Series(pd.period_range("2000-01-01", periods=10, freq="D"))
+#     pd_left, pd_right = pd_series[2], pd_series[7]
+#     expected = (pd_series >= pd_left) & (pd_series <= pd_right)
+#     series = Series(pd_series, chunk_size=5)
+#     left, right = series[2], series[7]
+#     result = series.between(left, right).execute().fetch()
+#     pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_inclusive_string(setup):
+    pd_series = pd.Series(pd.date_range("1/1/2000", periods=10))
+    pd_left, pd_right = pd_series[2], pd_series[7]
+    expected = (pd_series >= pd_left) & (pd_series <= pd_right)
+
+    series = Series(pd_series, chunk_size=5)
+    left, right = series[2], series[7]
+    result = series.between(left, right, inclusive="both").execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+    result = series.between(left, right, inclusive="left").execute().fetch()
+    expected = (pd_series >= pd_left) & (pd_series < pd_right)
+    pd.testing.assert_series_equal(result, expected)
+
+    result = series.between(left, right, inclusive="right").execute().fetch()
+    expected = (pd_series > pd_left) & (pd_series <= pd_right)
+    pd.testing.assert_series_equal(result, expected)
+
+    result = series.between(left, right, inclusive="neither").execute().fetch()
+    expected = (pd_series > pd_left) & (pd_series < pd_right)
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_error_args(setup):
+    series = Series(pd.date_range("1/1/2000", periods=10), chunk_size=5)
+    left, right = series[2], series[7]
+
+    value_error_msg = (
+        "Inclusive has to be either string of 'both',"
+        "'left', 'right', or 'neither'."
+    )
+
+    with pytest.raises(ValueError, match=value_error_msg):
+        series = Series(pd.date_range("1/1/2000", periods=10), chunk_size=5)
+        series.between(left, right, inclusive="yes").execute().fetch()
+
+
+def test_between_inclusive_warning(setup):
+    pd_series = pd.Series(pd.date_range("1/1/2000", periods=10))
+    pd_left, pd_right = pd_series[2], pd_series[7]
+    series = Series(pd_series, chunk_size=5)
+    left, right = series[2], series[7]
+    with pd.testing.assert_produces_warning(FutureWarning):
+        result = series.between(left, right, inclusive=False).execute().fetch()
+        expected = (pd_series > pd_left) & (pd_series < pd_right)
```

Comment:
Does not need to execute. Just test if a warning is produced.

Reply:
So I need to just remove lines 406 and 407?

Reply:
Guess so, the warning is gonna produced even without `execute`.

Reply:
done